contracts/ETHRegistrarController.sol

```diff
@@ -123,20 +140,38 @@ contract ETHRegistrarController is Ownable {
         if(msg.value > cost) {
             msg.sender.transfer(msg.value - cost);
         }
+
+        _sendReferralFee(referrer, cost);
     }
 
     function renew(string calldata name, uint duration) external payable {
-        uint cost = rentPrice(name, duration);
-        require(msg.value >= cost);
+        renewWithReferrer(name, duration, address(0));
+    }
 
-        bytes32 label = keccak256(bytes(name));
-        uint expires = base.renew(uint256(label), duration);
+    function renewWithReferrer(string memory name, uint duration, address payable referrer) public payable {
+        uint cost = _doRenew(name, duration);
+        require(msg.value >= cost);
 
+        // Refund any extra
         if(msg.value > cost) {
             msg.sender.transfer(msg.value - cost);
         }
 
-        emit NameRenewed(name, label, cost, expires);
+        _sendReferralFee(referrer, cost);
+    }
+    function renewAll(string[] calldata names, uint duration, address payable referrer) external payable {
```

Comment:
While integrating with `ens-app` I fond it inconsistent that `renewAll` requires referrer while `renew` is not.
Better to have `renewAll` and `renewAllWithReferrer`.

Reply:
I considered that, but there seemed little point introducing a new function that's redundant. I did this elsewhere because we need to maintain the old ABI for backwards compatibility.